src/NUnitConsole/nunit3-console/ConsoleRunner.cs

```diff
@@ -433,11 +435,51 @@ public static TestPackage MakeTestPackage(ConsoleOptions options)
                 package.AddSetting(FrameworkPackageSettings.DefaultTestNamePattern, options.DefaultTestNamePattern);
 
             if (options.TestParameters != null)
-                package.AddSetting(FrameworkPackageSettings.TestParameters, options.TestParameters);
+            {
+                var dictionary = new Dictionary<string, string>();
+
+                // Exactly what the framework used to do.
+                // Now can be independently changed without breaking backwards compatibility with old frameworks.
+                foreach (var param in options.TestParameters.Split(';'))
```

Comment:
@jnm2 In that case, change the comment to make sense long after this issue has merged. It's only at the present time that it's relevant to know what the framework used to do. If the code is not sufficiently self-explanatory, explain why it does what it does. The comment is also misleading in that the framework is going to continue splitting on unescaped semicolons for the foreseeable future.